cpp/benchmarks/synchronization/synchronization.hpp

```diff
@@ -0,0 +1,35 @@
+// Google Benchmark library
+#include <benchmark/benchmark.h>
+
+#include <tests/utilities/cudf_test_utils.cuh>
+
+class cuda_event_timer {
+ 
+public:
+  
+  // The c'tor clears the L2$ by cudaMemset a buffer of L2$ size
+  // and start the timer. 
+  cuda_event_timer(benchmark::State& state, cudaStream_t stream_ = 0);  
```

Comment:
This looks great now.

Just add Doxygen documentation that's inline with the example here: https://github.com/rapidsai/cudf/blob/branch-0.9/cpp/src/examples/documentation_example.cpp

I think it would be useful to include a description and example of what it looks like to use this `cuda_event_timer` class when writing a new Benchmark.